storage/allocator.go

```diff
@@ -81,6 +93,120 @@ func getUsedNodes(existing []proto.Replica) map[proto.NodeID]struct{} {
 	return usedNodes
 }
 
+// findDeadReplicas returns a list of nay replicas in the given RangeDescriptor
+// which are located on a dead store.
+func (a *allocator) findDeadReplicas(desc *proto.RangeDescriptor) []proto.Replica {
+	var deadReplicas []proto.Replica
+	// TODO(mrtracy): Pull in Bram's storePool changes to support this.
+	/*
+		for _, repl := range desc.Replicas {
+			if a.storePool.getStoreDetail(repl.StoreID).dead {
+				deadReplicas = append(deadReplicas, repl)
+			}
+		}
+		return deadReplicas
+	*/
+	return deadReplicas
+}
+
+// rangeAvailability computes some availability statistics about a range that
+// has dead replicas.
+func (a *allocator) computeRangeAvailability(desc *proto.RangeDescriptor, dead []proto.Replica) (bool, int) {
+	quorum := (len(desc.Replicas) / 2) + 1
+	liveReplicas := len(desc.Replicas) - len(dead)
+	if liveReplicas < quorum {
+		return false, 0
+	}
+	return true, quorum - liveReplicas
+}
+
+// requiresRepair determines if the supplied range, as governed by the supplied
+// zone configuration, requires any sort of repair action. It returns
+// information about whether or not the range *can* be repaired, if the range
+// *should* be repaired, and the relative priority of the required repair (if
+// one is required).
+func (a *allocator) requiresRepair(zone config.ZoneConfig, desc *proto.RangeDescriptor) (
+	canRepair bool, needsRepair bool, priority float64) {
+	// Search for dead replicas first.
+	deadReplicas := a.findDeadReplicas(desc)
+	if len(deadReplicas) > 0 {
+		isAvailable, priority := a.computeRangeAvailability(desc, deadReplicas)
+		// Return immediately if range has so many dead replicas that it is
+		// unavailable.
+		if !isAvailable {
+			return false, true, 0
+		}
+
+		// Removal of dead replicas is the highest priority.
+		return true, true, float64(300 + priority)
```

Comment:
can you change the 300 to a constant?  And go nuts, make it 
1000 for repair
700 for add
500 for subtract
0 for rebalance